### PR TITLE
[ParticleMechanicsApplication]/fix search condition geometry update 

### PR DESCRIPTION
**Pull request by VeronikaSinger**
Hi,

the geometry of the background element of the condition was not updated in the search. Therefore moving particle BC did not work. This PR fixes this issue and also added a test case to detect that error in the future.

### CODE REVIEW

**Comment by VeronikaSinger** [comment]
@tteschemacher can you take a look please?

File: applications/ParticleMechanicsApplication/tests/test_search_mpm_particle_condition.py [hunk 1]
```diff
@@ -0,0 +1,224 @@
+from __future__ import print_function, absolute_import, division
```
Comment (manuelmessmer) [inline applications/ParticleMechanicsApplication/tests/test_search_mpm_particle_condition.py]: I think you can remove this line.

**Review by manuelmessmer — DISMISSED** [review]
Hi @VeronikaSinger, looks good to me. Will approve to put some pressure on @tteschemacher ! ;)

File: applications/ParticleMechanicsApplication/tests/test_search_mpm_particle_condition.py (hunk 1 above)
Comment (philbucher) [reply applications/ParticleMechanicsApplication/tests/test_search_mpm_particle_condition.py]: yes, we no longer support python2

File: applications/ParticleMechanicsApplication/custom_utilities/mpm_particle_generator_utility.cpp [hunk 2]
```diff
@@ -439,6 +441,8 @@ namespace MPMParticleGeneratorUtility
                                 p_condition->SetValuesOnIntegrationPoints(MPC_IMPOSED_VELOCITY, { mpc_imposed_velocity }, process_info);
                                 p_condition->SetValuesOnIntegrationPoints(MPC_ACCELERATION, { mpc_acceleration }, process_info);
                                 p_condition->SetValuesOnIntegrationPoints(MPC_IMPOSED_ACCELERATION, { mpc_imposed_acceleration }, process_info);
+                                // Mark as boundary condition
+                                p_condition->Set(BOUNDARY, true);
```
Comment (tteschemacher) [inline applications/ParticleMechanicsApplication/custom_utilities/mpm_particle_generator_utility.cpp:445]: Where do you specifically use this?

File: applications/ParticleMechanicsApplication/custom_utilities/mpm_search_element_utility.h [hunk 3]
```diff
@@ -368,7 +371,13 @@ namespace MPMSearchElementUtility
                     bool is_found = SearchStructure.FindPointOnMesh(xg[0], N, pelem, result_begin, MaxNumberOfResults, Tolerance);
 
                     if (is_found == true) {
-                        condition_itr->GetGeometry() = pelem->GetGeometry();
+                        auto p_quadrature_point_geometry = condition_itr->pGetGeometry();
```
Comment (tteschemacher) [inline applications/ParticleMechanicsApplication/custom_utilities/mpm_search_element_utility.h:374]: Is that for lines in 2D?

In either scenario, you would need to update the tangents, too. I think we need to do this before the reset, though.

File: applications/ParticleMechanicsApplication/custom_utilities/mpm_particle_generator_utility.cpp (hunk 2 above)
Comment (VeronikaSinger) [reply applications/ParticleMechanicsApplication/custom_utilities/mpm_particle_generator_utility.cpp:445]: it is a ckeck in the search function - only boundary particles are searched not grid based conditions

File: applications/ParticleMechanicsApplication/custom_utilities/mpm_search_element_utility.h (hunk 3 above)
Comment (VeronikaSinger) [reply applications/ParticleMechanicsApplication/custom_utilities/mpm_search_element_utility.h:374]: at the moment those conditions are only points and if needed the normal vectors are updated from outside (if a contact condition is considered). If I understand correctly you are suggesting to update the normal/tangents according to the displacement of the background grid? I think that should be done in a differnt pr.

File: applications/ParticleMechanicsApplication/custom_utilities/mpm_particle_generator_utility.cpp (hunk 2 above)
Comment (tteschemacher) [reply applications/ParticleMechanicsApplication/custom_utilities/mpm_particle_generator_utility.cpp:445]: Ok. Maybe the name is not really perfect, considering that those could be also surface loads. Anyways, that would be something else.

File: applications/ParticleMechanicsApplication/custom_utilities/mpm_search_element_utility.h (hunk 3 above)
Comment (tteschemacher) [reply applications/ParticleMechanicsApplication/custom_utilities/mpm_search_element_utility.h:374]: Ok. I'm fine with that.

In either way, that was blocked for a long time. But now it is merged, so we can address that as soon as we need it.

File: applications/ParticleMechanicsApplication/tests/test_search_mpm_particle_condition.py (hunk 1 above)
Comment (tteschemacher) [inline applications/ParticleMechanicsApplication/tests/test_search_mpm_particle_condition.py]:
```suggestion
```

**Comment by tteschemacher** [comment]
Just remove that line as suggested. And I'd be absolutely fine to merge it.

Thanks a lot for the updates and efforts!

**Comment by VeronikaSinger** [comment]
@tteschemacher thanks a lot for your review

**Review by manuelmessmer — APPROVED** [review]

**Pull request merged**